### ISSUE TITLE
QCBXs are slow/unusable after adding several -to-many related records to a table

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3373]
https://herbrbge33123-v7810-prerelease.test.specifysystems.org/specify/view/collectingevent/389493/

**Steps to Recreate:**
1. See the above Collecting Event with 9 collectors.
2. Expand all of the collectors
3. Add a new one
4. Try to type name
5. See performance issues

It took a couple of different names/entries to recreate it, but once it happened, it stopped responding.

https://user-images.githubusercontent.com/37256050/233151909-859af172-e93b-416b-8357-c238cb2d1db8.mov

This issue looks like it will affect more than just the agent QCBX in the collectors subform, but this should be investigted.

**Reported By:** Robyn at RBGE

**Comment by grantfitzsimmons** [comment]
**Reported By:** Heather Cole at AAFC-AAC

>I was updating an existing record (to add collectors). I had four collectors to add, so I clicked the ‘+’ button four times, and it created 4 spots to add agents. For the first one, it behaved as I expected, with auto-complete recommendations starting as I typed, and let me select the agent I was looking for (e.g. Rosendahl, C.O.; first image). BUT THEN, when I tried to go to the next one (‘butters’; 2nd image), there was no auto-complete, or recommendations, and I had to use the search function (slow!) to search last name (which was in the system; 3rd image).
>
>Maybe I am supposed to ‘add, then fill’, then ‘add, then fill’ (instead of add, add, fill, fill)???
>
>Anyhoo, just thought I’d share, as a user, I would like to see all the additional agent fields search/autocomplete as I type (not just the first one).
>
> This doesn’t slow me down too much, or stop me from doing anything, just thought I’d pass it along, it can be potentially confusing if the same field “behaves different” in different situations.

<img width="227" alt="Screenshot 2023-08-04 at 11 50 19 AM" src="https://github.com/specify/specify7/assets/37256050/fe2913ea-e11c-443f-8d9e-ed1af32654ab">

<img width="583" alt="Screenshot 2023-08-04 at 11 50 27 AM" src="https://github.com/specify/specify7/assets/37256050/d6435310-a9c5-4657-a37d-e3cd4bd9a9db">

**Comment by maxxxxxdlp** [comment]
Related: https://github.com/specify/specify7/issues/2136
Will be possibly solved by https://github.com/specify/specify7/issues/1761

**Comment by grantfitzsimmons** [comment]
Reported again in Preparations by Eric Metz at Montana:
> Additionally, when we are doing preparations if we add more than 3 preparations to a single collection object entry, we can no longer use that drop list. It does not pop up. You can type in the drop box, but you cannot expand it.

**Comment by grantfitzsimmons** [comment]
Experienced again in `v7.9-dev` in my testing. This should be solved ASAP.

**Comment by grantfitzsimmons** [comment]
> We've had a user come across a weird issue, when adding multiple traits to a new collection object the auto-suggest on the agent stops working after the 2nd trait addition:
> 
> ![image](https://github.com/specify/specify7/assets/37256050/03ef5218-0f9a-44be-ba1d-dbcea96a8719)
> 
> They can still search for an agent using the search button, but have you seen this issue where the auto-suggest stops working?

**Reported by:** CSIRO on [Asana](https://app.asana.com/0/1201972579963460/1205789801005598/f)

**Comment by CarolineDenis** [comment]
The list is present actually, it is though for some reasons I am trying to find, displayed outside the frame. It's the same issue as: https://github.com/specify/specify7/issues/3458#issuecomment-1626188723

**Comment by CarolineDenis** [comment]
@grantfitzsimmons can we close this issue or the #3458 and paste the comments to the other one?

**Comment by grantfitzsimmons** [comment]
Yes, we can merge #3458 into this issue. It appears to be the same underlying issue.

**Comment by CarolineDenis** [comment]
Duplicate of https://github.com/specify/specify7/issues/3458

**Comment by grantfitzsimmons** [comment]
I suggest merging issue #3458 into this issue, rather than the other way around.

**Comment by emenslin** [comment]
Can recreate in edge (7.9.6)

**Comment by lexiclevenger** [comment]
The case from #3458 has also been reported by Christian Isenvadh at GMNH with the Storage field under Preparations:

> We found a strange issue. When opening multiple objects from a query, we are not able to search by typing in the storage combobox. This is no problem when opening the object separately. See enclosed images. The behavior is the same across different users, disciplines and environments.
